src/sentry/integrations/utils/code_mapping.py

```diff
@@ -342,12 +325,7 @@ def _generate_code_mapping_from_tree(
         if len(matched_files) != 1:
             return []
 
-        stacktrace_root = f"{frame_filename.root}/"
-        source_path = _get_code_mapping_source_path(matched_files[0], frame_filename)
-        if frame_filename.frame_type() != "packaged":
-            stacktrace_root, source_path = self._normalized_stack_and_source_roots(
-                stacktrace_root, source_path
-            )
+        stacktrace_root, source_path = find_roots(frame_filename.raw_path, matched_files[0])
```

Comment:
This is where root derivation occurs for automatic code mappings - replace the previous logic with `find_roots`